### PR TITLE
Личный проект: пока все дома

### PR DESCRIPTION
**Pull request by Laptenok-vita**
---
:mortar_board: [Личный проект: пока все дома](https://up.htmlacademy.ru/javascript/14/user/419117/tasks/13)

:boom: https://htmlacademy-javascript.github.io/419117-keksobooking/2/

### CODE REVIEW

File: js/map.js [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+var AVATAR_AMOUNT = ['01', '02', '03', '04', '05', '06', '07', '08'];
+var TITLE = ['Большая уютная квартира', 'Маленькая неуютная квартира', 'Огромный прекрасный дворец', 'Маленький ужасный дворец', 'Красивый гостевой домик', 'Некрасивый негостеприимный домик', 'Уютное бунгало далеко от моря', 'Неуютное бунгало по колено в воде'];
+var TYPES_HOUSES = ['palace', 'flat', 'house', 'bungalo'];
+var TIME_CHECKIN = ['12:00', '13:00', '14:00'];
+var TIME_CHECKOUT = ['12:00', '13:00', '14:00'];
+var FEATURES = ['wifi', 'dishwasher', 'parking', 'washer', 'elevator', 'conditioner'];
+var PHOTOS = ['http://o0.github.io/assets/images/tokyo/hotel1.jpg', 'http://o0.github.io/assets/images/tokyo/hotel2.jpg', 'http://o0.github.io/assets/images/tokyo/hotel3.jpg'];
+
+// Функция генерирует случайное число в заданом диапазоне
+var ARRAY_NUMBERS =
```
Comment (rkhaslarov) [inline js/map.js]: Why do u use the object here? Maybe it's better to use simple function?

File: js/map.js [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+  {
+    createNumber: function (min, max) {
+      return Math.floor(Math.random() * (max - min) + min);
+    }
+  };
+
+  // Функция перебирает массив и возвращает случайное значение
+var getRandomNumber = function shuffleArray(array) {
```
Comment (rkhaslarov) [inline js/map.js]: remove the name from right hand side, I mean shuffleArray and change please getRandomNumber to shuffleArray

File: js/map.js [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+// Нахожу элемент, куда буду вставлять разметку
+similarListElement = document.querySelector('.map');
+// Нахожу тег template в разметке
+similarMapElementTemplate = document.querySelector('template')
+// Нахожу (с помощью дом-узла DocumentFragment, который называется content) тег, который мне надо скопировать
+    .content
+    .querySelector('.map__card');
+
+var renderMapElement = function (mapArray) {
+  var MapElement = similarMapElementTemplate.cloneNode(true);
```
Comment (rkhaslarov) [inline js/map.js]: camelCase please

File: js/map.js [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+
+var fragment = document.createDocumentFragment();
+for (var i = 0; i < 8; i++) {
+  similarRentAds[i] = {
+    'author': {
+      avatar: 'img/avatars/user' + getRandomNumber(AVATAR_AMOUNT) + '.png'
+    },
+    'offer': {
+      'title': getRandomNumber(TITLE),
+      'address': ARRAY_NUMBERS.createNumber(300, 900) + ', ' + ARRAY_NUMBERS.createNumber(130, 630),
```
Comment (rkhaslarov) [inline js/map.js]: x and y have to be the same as in location, use variables for this case

File: js/map.js [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+  MapElement.querySelector('.popup__avatar').textContent = mapArray.author.avatar;
+
+  return MapElement;
+};
+
+var fragment = document.createDocumentFragment();
+for (var i = 0; i < 8; i++) {
+  similarRentAds[i] = {
+    'author': {
+      avatar: 'img/avatars/user' + getRandomNumber(AVATAR_AMOUNT) + '.png'
```
Comment (rkhaslarov) [inline js/map.js]: No need to randomize number, use just i as a number of image

File: js/map.js [hunk 6]
```diff
@@ -0,0 +1,101 @@
...
+var TYPES_HOUSES = ['palace', 'flat', 'house', 'bungalo'];
+var TIME_CHECKIN = ['12:00', '13:00', '14:00'];
+var TIME_CHECKOUT = ['12:00', '13:00', '14:00'];
+var FEATURES = ['wifi', 'dishwasher', 'parking', 'washer', 'elevator', 'conditioner'];
+var PHOTOS = ['http://o0.github.io/assets/images/tokyo/hotel1.jpg', 'http://o0.github.io/assets/images/tokyo/hotel2.jpg', 'http://o0.github.io/assets/images/tokyo/hotel3.jpg'];
+
+// Функция генерирует случайное число в заданом диапазоне
+var ARRAY_NUMBERS =
+  {
+    createNumber: function (min, max) {
```
Comment (rkhaslarov) [inline js/map.js]: This is the getRandomNumber function

File: js/map.js [hunk 7]
```diff
@@ -0,0 +1,101 @@
...
+  MapElement.querySelector('.popup__description').textContent = mapArray.offer.description;
+  MapElement.querySelector('.popup__photos').src = mapArray.offer.photos;
+  MapElement.querySelector('.popup__avatar').textContent = mapArray.author.avatar;
+
+  return MapElement;
+};
+
+var fragment = document.createDocumentFragment();
+for (var i = 0; i < 8; i++) {
+  similarRentAds[i] = {
```
Comment (rkhaslarov) [inline js/map.js]: use variable here, and after that use push function of array

File: js/map.js [hunk 8]
```diff
@@ -0,0 +1,101 @@
...
+      'description': '',
+      'photos': getRandomNumber(PHOTOS)
+    },
+    'location': {
+      'x': ARRAY_NUMBERS.createNumber(300, 900),
+      'y': ARRAY_NUMBERS.createNumber(130, 630)
+    }
+  };
+
+  fragment.appendChild(renderMapElement(similarRentAds[i]));
```
Comment (rkhaslarov) [inline js/map.js]: pass variable instead of similarRentAds[i]

File: js/map.js [hunk 9]
```diff
@@ -0,0 +1,101 @@
...
+    .content
+    .querySelector('.map__pin');
+
+// Нахожу элемент, куда буду вставлять разметку
+similarListElement = document.querySelector('.map');
+// Нахожу тег template в разметке
+similarMapElementTemplate = document.querySelector('template')
+// Нахожу (с помощью дом-узла DocumentFragment, который называется content) тег, который мне надо скопировать
+    .content
+    .querySelector('.map__card');
```
Comment (rkhaslarov) [inline js/map.js]: It's completely wrong, you use the same variable for card and pin templates, why? The correct resolution should be like I wrote below, also I don't see any benefits of such naming that used here, no need to dream up completely new names like similar bla-bla-bla, just use everything that you have currently:
var mapPins = document.querySelector('.map__pins');
var pinTemplate = document.querySelector('template')
+// Нахожу (с помощью дом-узла DocumentFragment, который называется content) тег, который мне надо скопировать
+    .content
+    .querySelector('.map__pin');

var map = document.querySelector('.map');
var mapTemplate = document.querySelector('template')
+// Нахожу (с помощью дом-узла DocumentFragment, который называется content) тег, который мне надо скопировать
+    .content
+    .querySelector('.map__card');

File: js/map.js [hunk 10]
```diff
@@ -0,0 +1,188 @@
+'use strict';
+
+var AVATAR_AMOUNT = [
```
Comment (rkhaslarov) [inline js/map.js]: Should be plural

File: js/map.js [hunk 11]
```diff
@@ -0,0 +1,188 @@
...
+  'img/avatars/user02.png',
+  'img/avatars/user03.png',
+  'img/avatars/user04.png',
+  'img/avatars/user05.png',
+  'img/avatars/user06.png',
+  'img/avatars/user07.png',
+  'img/avatars/user08.png'
+];
+
+var TITLE = [
```
Comment (rkhaslarov) [inline js/map.js]: Should be plural

File: js/map.js [hunk 12]
```diff
@@ -0,0 +1,188 @@
...
+  'house',
+  'bungalo'
+];
+
+var TYPES_HOUSES_RUS = [
+  'Дворец',
+  'Квартира',
+  'Дом',
+  'Бунгало'
+];
```
Comment (rkhaslarov) [inline js/map.js]: Use objects instead of these two arrays

File: js/map.js [hunk 13]
```diff
@@ -0,0 +1,188 @@
...
+    var j = Math.floor(Math.random() * (i + 1));
+    var temp = array[i];
+    array[i] = array[j];
+    array[j] = temp;
+  }
+  return temp;
+};
+
+var unshuffleArray = function (array) {
+  var i = Math.floor(Math.random() * array.length);
```
Comment (rkhaslarov) [inline js/map.js]: Unnecessary variable and weird name for function, it's better to name it like getRandomValueFromArray, don't afraid of long names of functions. It brings readability and consistency

File: js/map.js [hunk 14]
```diff
@@ -0,0 +1,188 @@
...
+  return array[i];
+};
+
+var getRandomLength = function (array) {
```
Comment (rkhaslarov) [inline js/map.js]: Should be renamed to getRandomValuesFromArray

File: js/map.js [hunk 15]
```diff
@@ -0,0 +1,188 @@
...
+
+var getRandomPhotos = function (array) {
+  var cloneArray = array.slice();
+  for (var i = cloneArray.length - 1; i >= 0; i--) {
+    var j = Math.floor(Math.random() * (i + 1));
+    var temp = cloneArray[i];
+    cloneArray[i] = cloneArray[j];
+    cloneArray[j] = temp;
+  }
+  return cloneArray;
```
Comment (rkhaslarov) [inline js/map.js]: Why do you need for this function? You have the same thing in shuffle array

File: js/map.js [hunk 16]
```diff
@@ -0,0 +1,188 @@
...
+
+  return mapCard;
+};
+
+var adsArray = [];
+var fragmentPin = document.createDocumentFragment();
+var fragmentMapAd = document.createDocumentFragment();
+
+for (var i = 0; i < 8; i++) {
+  var CoordinateX = createNumber(300, 900) - PIN_WIDTH;
```
Comment (rkhaslarov) [inline js/map.js]: Firstly, I want to note that you should use camelCase notation. Also why it can't be simply x and y?

File: js/map.js [hunk 17]
```diff
@@ -0,0 +1,188 @@
...
+      'description': '',
+      'photos': getRandomPhotos(PHOTOS)
+    },
+    'location': {
+      'x': CoordinateX,
+      'y': CoordinateY
+    }
+  };
+
+  adsArray.push(rentAd);
```
Comment (rkhaslarov) [inline js/map.js]: Move all logic related to creating an object and pushing it to array to the separate function

File: js/map.js [hunk 18]
```diff
@@ -0,0 +1,188 @@
...
+  mapCard.querySelector('.popup__description').textContent = mapAdArray.offer.description;
+  mapCard.querySelector('.popup__photo').src = mapAdArray.offer.photos[0];
+  mapCard.appendChild(mapCard.querySelector('.popup__photo').cloneNode(true)).src = mapAdArray.offer.photos[1];
+  mapCard.appendChild(mapCard.querySelector('.popup__photo').cloneNode(true)).src = mapAdArray.offer.photos[2];
+  mapCard.querySelector('img').src = mapAdArray.author.avatar;
+
+  return mapCard;
+};
+
+var adsArray = [];
```
Comment (rkhaslarov) [inline js/map.js]: just ads

File: js/map.js (hunk 15 above)
Comment (Laptenok-vita) [reply js/map.js]: This function returns an array of strings in a random order according to clause 5.9 of the job. The shuffle array function returns one random array string.

File: js/map.js [hunk 19]
```diff
@@ -0,0 +1,192 @@
...
+  'Уютное бунгало далеко от моря',
+  'Неуютное бунгало по колено в воде'
+];
+
+var TYPES_HOUSES = [
+  {'palace': 'Дворец'},
+  {'flat': 'Квартира'},
+  {'house': 'Дом'},
+  {'bungalo': 'Бунгало'}
+];
```
Comment (rkhaslarov) [inline js/map.js]: Oh, use single object for all types: 
{
'palace': 'Дворец'
'flat': 'Квартира',
...
}

File: js/map.js [hunk 20]
```diff
@@ -0,0 +1,192 @@
...
+var mapAd = document.querySelector('.map');
+var mapAdTemplate = document.querySelector('template')
+  .content
+  .querySelector('.map__card');
+
+var createNumber = function (min, max) {
+  return Math.floor(Math.random() * (max - min) + min);
+};
+
+var shuffleArray = function (array) {
```
Comment (rkhaslarov) [inline js/map.js]: Rename it to getRandomValueFromArray, and try next piece of code:

 return array[Math.floor(Math.random() * array.length)];

File: js/map.js [hunk 21]
```diff
@@ -0,0 +1,192 @@
...
+  for (var i = array.length - 1; i >= 0; i--) {
+    var j = Math.floor(Math.random() * (i + 1));
+    var temp = array[i];
+    array[i] = array[j];
+    array[j] = temp;
+  }
+  return temp;
+};
+
+var getRandomPhotos = function (array) {
```
Comment (rkhaslarov) [inline js/map.js]: Rename to shuffleArray

File: js/map.js [hunk 22]
```diff
@@ -0,0 +1,192 @@
...
+
+var indexArray = TYPES_HOUSES[createNumber(0, TYPES_HOUSES.length)];
+for (var key in indexArray) {
+  var houseType = indexArray[key];
+}
+
+var createAds = function () {
+  var rentAd = {
+    'author': {
+      'avatar': AVATARS_AMOUNT[getRandomIndex(AVATARS_AMOUNT)]
```
Comment (rkhaslarov) [inline js/map.js]: Avatars shouldn't be random use pass index of loop here and just use it

File: js/map.js [hunk 23]
```diff
@@ -0,0 +1,192 @@
...
+      'y': y
+    }
+  };
+
+  ads.push(rentAd);
+};
+
+for (var i = 0; i < 8; i++) {
+  var x = createNumber(300, 900) - PIN_WIDTH;
+  var y = createNumber(130, 630) - PIN_HEIGHT;
```
Comment (rkhaslarov) [inline js/map.js]: No need to deduct pin size, also move x and y to createAd function

File: js/map.js [hunk 24]
```diff
@@ -0,0 +1,192 @@
...
+  createAds();
```
Comment (rkhaslarov) [inline js/map.js]: ads.push(createAds(i));

File: js/map.js [hunk 25]
```diff
@@ -0,0 +1,192 @@
...
+      'description': '',
+      'photos': getRandomPhotos(PHOTOS)
+    },
+    'location': {
+      'x': x,
+      'y': y
+    }
+  };
+
+  ads.push(rentAd);
```
Comment (rkhaslarov) [inline js/map.js]: move this below

File: js/map.js [hunk 26]
```diff
@@ -0,0 +1,192 @@
...
+  {'bungalo': 'Бунгало'}
+];
+
+var TIME_CHECKIN = [
+  '12:00',
+  '13:00',
+  '14:00'
+];
+
+var TIME_CHECKOUT = [
```
Comment (rkhaslarov) [inline js/map.js]: CHECKOUT_DATES

File: js/map.js [hunk 27]
```diff
@@ -0,0 +1,192 @@
...
+];
+
+var TYPES_HOUSES = [
+  {'palace': 'Дворец'},
+  {'flat': 'Квартира'},
+  {'house': 'Дом'},
+  {'bungalo': 'Бунгало'}
+];
+
+var TIME_CHECKIN = [
```
Comment (rkhaslarov) [inline js/map.js]: CHECKIN_DATES

File: js/map.js [hunk 28]
```diff
@@ -0,0 +1,192 @@
+'use strict';
+
+var AVATARS_AMOUNT = [
```
Comment (rkhaslarov) [inline js/map.js]: AVATARS

File: js/map.js [hunk 29]
```diff
@@ -0,0 +1,185 @@
+'use strict';
+
+var ADS_AMOUNT = 8;
+
+var TITLES = [
```
Comment (rkhaslarov) [inline js/map.js]: AD_TITLES

File: js/map.js [hunk 30]
```diff
@@ -0,0 +1,185 @@
...
+  'Маленькая неуютная квартира',
+  'Огромный прекрасный дворец',
+  'Маленький ужасный дворец',
+  'Красивый гостевой домик',
+  'Некрасивый негостеприимный домик',
+  'Уютное бунгало далеко от моря',
+  'Неуютное бунгало по колено в воде'
+];
+
+var TYPES_HOUSES =
```
Comment (rkhaslarov) [inline js/map.js]: HOUSE_TYPES

File: js/map.js [hunk 31]
```diff
@@ -0,0 +1,185 @@
...
+var PHOTOS = [
+  'http://o0.github.io/assets/images/tokyo/hotel1.jpg',
+  'http://o0.github.io/assets/images/tokyo/hotel2.jpg',
+  'http://o0.github.io/assets/images/tokyo/hotel3.jpg'
+];
+
+var PIN_WIDTH = 40;
+var PIN_HEIGHT = 40;
+
+var createNumber = function (min, max) {
```
Comment (rkhaslarov) [inline js/map.js]: getRandomNumberInRange

File: js/map.js [hunk 32]
```diff
@@ -0,0 +1,185 @@
...
+  return array;
+};
+
+var getRandomLength = function (array) {
+  var cloneArray = array.slice();
+  cloneArray.length = Math.round(Math.random() * array.length);
+  return cloneArray;
+};
+
+var getRandomIndex = function (array) {
```
Comment (rkhaslarov) [inline js/map.js]: Unnecessary function

File: js/map.js [hunk 33]
```diff
@@ -0,0 +1,185 @@
...
+var createAd = function (i) {
+  var x = createNumber(300, 900) - PIN_WIDTH;
+  var y = createNumber(130, 630) - PIN_HEIGHT;
+
+  var rentAd = {
+    'author': {
+      'avatar': 'img/avatars/user0' + (i + 1) + '.png'
+    },
+    'offer': {
+      'title': TITLES[getRandomIndex(TITLES)],
```
Comment (rkhaslarov) [inline js/map.js]: Use i instead of getRandomIndex

File: js/map.js [hunk 34]
```diff
@@ -0,0 +1,185 @@
...
+    var temp = array[i];
+    array[i] = array[j];
+    array[j] = temp;
+  }
+  return array;
+};
+
+var getRandomLength = function (array) {
+  var cloneArray = array.slice();
+  cloneArray.length = Math.round(Math.random() * array.length);
```
Comment (rkhaslarov) [inline js/map.js]: Use shuffle function

File: js/map.js [hunk 35]
```diff
@@ -0,0 +1,185 @@
...
+var pinBefore = document.querySelector('.map__filters-container');
+
+var mapAd = document.querySelector('.map');
+var mapAdTemplate = document.querySelector('template')
+  .content
+  .querySelector('.map__card');
+
+for (var key in TYPES_HOUSES) {
+  var houseType = TYPES_HOUSES[key];
+}
```
Comment (rkhaslarov) [inline js/map.js]: Remove this loop

File: js/map.js [hunk 36]
```diff
@@ -0,0 +1,185 @@
...
+
+  pin.style.left = pinArray.location.x + 'px';
+  pin.style.top = pinArray.location.y + 'px';
+  pin.querySelector('img').src = pinArray.author.avatar;
+  pin.querySelector('img').alt = pinArray.offer.title;
+
+  return pin;
+};
+
+var renderMapAd = function (mapAdArray) {
```
Comment (rkhaslarov) [inline js/map.js]: renderMapCard - ad

File: js/map.js [hunk 37]
```diff
@@ -0,0 +1,185 @@
...
+  var mapCard = mapAdTemplate.cloneNode(true);
+
+  mapCard.querySelector('.popup__title').textContent = mapAdArray.offer.title;
+  mapCard.querySelector('.popup__text--address').textContent = mapAdArray.offer.address;
+  mapCard.querySelector('.popup__text--price').textContent = mapAdArray.offer.price + ' ₽/ночь';
+  mapCard.querySelector('.popup__type').textContent = mapAdArray.offer.type;
```
Comment (rkhaslarov) [inline js/map.js]: HOUSE_TYPES_DICT[ad.offer.type]

File: js/map.js [hunk 38]
```diff
@@ -0,0 +1,185 @@
...
+  mapCard.querySelector('.popup__text--capacity').textContent = mapAdArray.offer.rooms + ' комнаты для ' + mapAdArray.offer.guests + ' гостей';
+  mapCard.querySelector('.popup__text--time').textContent = 'Заезд после ' + mapAdArray.offer.checkin + ', выезд до ' + mapAdArray.offer.checkout;
+  mapCard.querySelector('.popup__features').textContent = mapAdArray.offer.features;
```
Comment (rkhaslarov) [inline js/map.js]: Rewrite features to use ul

File: js/map.js [hunk 39]
```diff
@@ -0,0 +1,185 @@
...
+  mapCard.querySelector('.popup__description').textContent = mapAdArray.offer.description;
+  mapCard.querySelector('.popup__photo').src = mapAdArray.offer.photos[0];
+  mapCard.appendChild(mapCard.querySelector('.popup__photo').cloneNode(true)).src = mapAdArray.offer.photos[1];
```
Comment (rkhaslarov) [inline js/map.js]: Rewrite photos too

File: js/map.js [hunk 40]
```diff
@@ -0,0 +1,185 @@
...
+  mapCard.appendChild(mapCard.querySelector('.popup__photo').cloneNode(true)).src = mapAdArray.offer.photos[2];
+  mapCard.querySelector('img').src = mapAdArray.author.avatar;
+
+  return mapCard;
+};
+
+var fragmentPin = document.createDocumentFragment();
+var fragmentMapAd = document.createDocumentFragment();
```
Comment (rkhaslarov) [inline js/map.js]: Create renderPins function

File: js/map.js [hunk 41]
```diff
@@ -0,0 +1,185 @@
...
+
+for (var i = 0; i < ADS_AMOUNT; i++) {
+  fragmentPin.appendChild(renderMapPin(ads[i]));
+}
+
+fragmentMapAd.appendChild(renderMapAd(ads[0]));
+
+mapPins.appendChild(fragmentPin);
```
Comment (rkhaslarov) [inline js/map.js]: mapPins.appendChild(renderPins());

File: js/map.js [hunk 42]
```diff
@@ -0,0 +1,194 @@
...
+var getRandomValuesFromArray = function (array) {
+  var cloneArray = shuffleArray(array);
+  cloneArray.length = getRandomNumberInRange(1, cloneArray.length);
+  return cloneArray;
+};
+
+var mapToggle = document.querySelector('.map');
+mapToggle.classList.remove('map--faded');
+
+var mapPins = document.querySelector('.map__pins');
```
Comment (rkhaslarov) [inline js/map.js]: Move all variable up to the file's beginning

File: js/map.js [hunk 43]
```diff
@@ -0,0 +1,194 @@
...
+var pinTemplate = document.querySelector('template')
+  .content
+  .querySelector('.map__pin');
+var pinBefore = document.querySelector('.map__filters-container');
+
+var mapAd = document.querySelector('.map');
+var AdTemplate = document.querySelector('template')
```
Comment (rkhaslarov) [inline js/map.js]: adTemplate - don't forget camelCase

File: js/map.js [hunk 44]
```diff
@@ -0,0 +1,194 @@
...
+
+  var rentAd = {
+    'author': {
+      'avatar': 'img/avatars/user0' + (i + 1) + '.png'
+    },
+    'offer': {
+      'title': AD_TITLES[i + 1],
+      'address': x + ', ' + y,
+      'price': getRandomNumberInRange(1000, 1000000),
+      'type': HOUSES_TYPES[getRandomNumberInRange(0, HOUSES_TYPES.length)],
```
Comment (rkhaslarov) [inline js/map.js]: Use getRandomValueFromArray

File: js/map.js [hunk 45]
```diff
@@ -0,0 +1,193 @@
...
+var FEATURES = [
+  'wifi',
+  'dishwasher',
+  'parking',
+  'washer',
+  'elevator',
+  'conditioner'
+];
+
+var PHOTOS = [
```
Comment (rkhaslarov) [inline js/map.js]: AD_PHOTOS

File: js/map.js [hunk 46]
```diff
@@ -0,0 +1,193 @@
...
+  '14:00'
+];
+
+var CHECKOUT_DATES = [
+  '12:00',
+  '13:00',
+  '14:00'
+];
+
+var FEATURES = [
```
Comment (rkhaslarov) [inline js/map.js]: AD_*

File: js/map.js [hunk 47]
```diff
@@ -0,0 +1,193 @@
...
+
+  var rentAd = {
+    'author': {
+      'avatar': 'img/avatars/user0' + (i + 1) + '.png'
+    },
+    'offer': {
+      'title': AD_TITLES[i + 1],
+      'address': x + ', ' + y,
+      'price': getRandomNumberInRange(1000, 1000000),
+      'type': HOUSES_TYPES[getRandomValueFromArray(0, HOUSES_TYPES.length)],
```
Comment (rkhaslarov) [inline js/map.js]: Regression, getRandomValueFromArray

File: js/map.js [hunk 48]
```diff
@@ -0,0 +1,193 @@
...
+  var ads = [];
+  for (var i = 0; i < ADS_AMOUNT; i++) {
+    ads[i] = createAd(i);
+  }
+  return ads;
+};
+
+var ads = createAds();
+
+var renderPin = function (pinArray) {
```
Comment (rkhaslarov) [inline js/map.js]: simply pin, it's not an array

**Pull request merged**